indico/core/db/sqlalchemy/principals.py

```diff
@@ -413,13 +413,29 @@ def get_emails(self):
         """Get a set of all unique emails associated with this principal.
 
         For users, this is just the primary email (or nothing for the system user).
-        For groups it is the primary email address of each group members who have
-        an Indico account.
+        For anything group-like it is the primary email address of each group member
+        who has an Indico account.
         """
         if self.type == PrincipalType.user and not self.user.is_system:
             return {self.user.email}
         elif self.type in (PrincipalType.local_group, PrincipalType.multipass_group):
             return {x.email for x in self.principal.get_members() if not x.is_system}
+        elif self.type in (PrincipalType.event_role, PrincipalType.category_role):
+            return {x.email for x in self.principal.members if not x.is_system}
+        return set()
+
+    def get_users(self):
```

Comment:
What do you think about making this generic (getting a principal's attribute)? Such as:
```
get_attribute(attribute, include_system=False): ...

get_attribute('email', include_system=True) // this would be emails
```

Could this be useful in the future?